src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1080,16 +1081,21 @@ private void writeLogTo(WriteMethod method, OutputStream os) throws IOException
 
     @Override public File getLogFile() {
         LOGGER.log(Level.WARNING, "Avoid calling getLogFile on " + this, new UnsupportedOperationException());
-        try {
-            File f = File.createTempFile("deprecated", ".log", getRootDir());
-            f.deleteOnExit();
-            try (OutputStream os = new FileOutputStream(f)) {
-                getLogText().writeRawLogTo(0, os);
+        File f = super.getLogFile();
+        if (LogStorage.of(asFlowExecutionOwner()) instanceof FileLogStorage) {
+            // FileLogStorage does write a file with the same name and the same format as before JEP-210, so accept it if it is there.
+        } else {
+            // Some other storage. Cache under the traditional name, but load from the defined source on each call.
+            try {
+                f.deleteOnExit();
```

Comment:
Hey, isn't `f` potentially the original log file, not a deprecated temp file that we're copying to/from?

Reply:
Which means we'd... delete the build log on exit?  Unless there's something I missed here?

Reply:
That would be the `FileLogStorage` branch. I.e., what is run if you do _not_ have the CloudWatch Logs plugin installed.

Reply:
Yeah, but in that case... isn't that the *actual* build log?

Reply:
Gotta say it: we really, really, really should have a testcase covering these scenarios.  Relying on code review alone to catch issues here is not a longterm path for success.

Reply:
In the case of `FileLogStorage`, yes that is the actual build log, as it would be for freestyle builds.

Reply:
🐛 So we're marking for Jenkins to... delete our build's actual logfile on exit. 🐛

Reply:
Ah, except... due to magic the legacy logs magically have become a FileLogStorage due to the other weirdness of the `LogStorage.of` call.  Which raises the question: what is this code addressing?

Reply:
Unless some plugin adds an alternative, `FileLogStorage` is in use. It does not matter if this build predates JEP-210.